src/games/citations.c

```diff
@@ -422,7 +429,7 @@ int igraph_citing_cited_type_game(igraph_t *graph, igraph_integer_t nodes,
     for (i = 0; i < nocats; i++) {
         long int type = (long int) VECTOR(*types)[0];
         if ( MATRIX(*pref, i, type) < 0) {
-            IGRAPH_ERROR("pref contains negative entries", IGRAPH_EINVAL);
+            IGRAPH_ERRORF("Preference matrix contains negative entry: %g.", IGRAPH_EINVAL, MATRIX(*pref, i, type));
```

Comment:
Not mandatory, but the approach I use when the very same error may be triggered from multiple places is to put the error handling after the `return` at the end of the function, put a label on it, and jump to it with `goto`. `goto` is not always evil :-)  See `igraph_cited_type_game`.

Reply:
Of course that wouldn't work here because we print the matrix element, so ignore my comment. Sorry.